ERCS/erc-7704.md

```diff
@@ -0,0 +1,172 @@
+
+## Specification
+
+Notes:
+The contract is designed to be compiled with a Solidity compiler version greater than or equal to 0.8.20 but less than 0.9.0.
+
+The smart contract realising this EIP mandate must fully incorporate all methods prescribed by [ERC-20](./eip-20.md), with the exception of transfer operations. Furthermore, exclusivity to implementing solely this EIP is required.
+ 
+## Rationale
+
+SRT serves as an extension to [ERC-20](./eip-20.md), [ERC-721](./eip-721.md), or [ERC-1155](./eip-1155.md), enabling seamless integration with any of these standards. It advocates for the deactivation of the transfer method to uphold clarity in ownership relations. Token transactions are exclusively conducted through minting and burning methods, facilitating uncomplicated permission management. Furthermore, reputation computation can be customized using bonding curves, with implementation guided by adherence to the SRT interface.
+
+## Backwards Compatibility
+
+This proposal is only partially compatible with [ERC-20](./eip-20.md) because it makes tokens non-transferable after the first transfer.
+
+**Implementations**
+
+## Reference Implementation
+
+```solidity
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.20;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract SRT20Minimal is ERC20 {
+    error TransferNotAllowed();
+
+    constructor() ERC20("Soul Resonance Token 20", "SRT20") {}
+
+    function _update(
+        address from,
+        address to,
+        uint256 value
+    ) internal virtual override {
+        if (from != address(0) && to != address(0)) {
+            revert TransferNotAllowed();
+        }
+        super._update(from, to, value);
+    }
+
+    function mint(address to, uint256 amount) public {
+        _mint(to, amount);
+    }
+
+    function burn(address from, uint256 amount) public {
+        _burn(from, amount);
+    }
+
+}
+```
+
+```solidity
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+contract SRT721Minimal is ERC721 {
+    error TransferNotAllowed();
+
+    constructor() ERC721("Soul Resonance Token 721", "SRT721") {}
+
+    function _update(
+        address to,
+        uint256 tokenId,
+        address auth
+    ) internal virtual override returns (address) {
+        address from = super._update(to, tokenId, auth);
+        if (from != address(0) && to != address(0)) {
+            revert TransferNotAllowed();
+        }
+        return from;
+    }
+
+    function safeMint(address to, uint256 tokenId) public {
+        _mint(to, tokenId);
+    }
+
+    function burn(uint256 tokenId) public {
+        _update(address(0), tokenId, msg.sender);
+    }
+
+}
+```
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code within the document must be CC0-1.0 licensed. You may move these to your assets directory if you'd like to keep them MIT.